### PR TITLE
feature(cli): Adding CLI cloud login and auth logic

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the base logic to support CLI cloud login and authentication

## Changes

- Adds `tracetest login` command
- Implements oauth logic for resources and the API

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: cli/cmd/resources.go [hunk 1]
```diff
@@ -271,6 +271,9 @@ func setupResources() {
 	extraHeaders := http.Header{}
 	extraHeaders.Set("x-client-id", analytics.ClientID())
 	extraHeaders.Set("x-source", "cli")
+	extraHeaders.Set("x-organization-id", cliConfig.OrganizationID)
```
Comment (xoscar) [inline cli/cmd/resources.go:274]: We are attaching the new headers as part of the request

File: cli/cmd/login_cmd.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	Long:    "Initializes the Tracetst Cloud Login process",
+	PreRun:  setupCommand(),
+	Run: WithResultHandler(func(_ *cobra.Command, _ []string) (string, error) {
+		ctx := context.Background()
+		onSuccess := func(token string, jwt string) {
+			fmt.Printf("✔ Successful Authenticated with token: %s and jwt: %s\n", token, jwt)
+			cliConfig.Token = token
+			cliConfig.Jwt = jwt
+			serverPath := ""
+			cliConfig.ServerPath = &serverPath
```
Comment (xoscar) [inline cli/cmd/login_cmd.go:24]: We have to remove the `/api` server prefix

File: cli/pkg/resourcemanager/client.go [hunk 3]
```diff
@@ -47,7 +47,7 @@ func NewHTTPClient(baseURL string, extraHeaders http.Header) *HTTPClient {
 }
 
 func (c HTTPClient) url(resourceName string, extra ...string) *url.URL {
-	urlStr := c.baseURL + path.Join("/api", resourceName, strings.Join(extra, "/"))
+	urlStr := c.baseURL + path.Join("/", resourceName, strings.Join(extra, "/"))
```
Comment (xoscar) [inline cli/pkg/resourcemanager/client.go:50]: The prefix will be part of the server path now

File: cli/cmd/login_cmd.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	"fmt"
+
+	"github.com/kubeshop/tracetest/cli/config"
+	"github.com/spf13/cobra"
+)
+
+var loginCmd = &cobra.Command{
+	GroupID: cmdGroupCloud.ID,
+	Use:     "login",
+	Short:   "Tracetst Cloud Login",
```
Comment (jorgeepc) [inline cli/cmd/login_cmd.go:14]:
```suggestion
	Short:   "Tracetest Cloud Login",
```

File: cli/cmd/login_cmd.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	Long:    "Initializes the Tracetst Cloud Login process",
```
Comment (jorgeepc) [inline cli/cmd/login_cmd.go:15]:
```suggestion
	Long:    "Initializes the Tracetest Cloud Login process",
```

File: cli/cmd/resources.go (hunk 1 above)
Comment (jorgeepc) [reply cli/cmd/resources.go:274]: So we are always going to send this headers, right? even for OSS?

**Review by jorgeepc — APPROVED** [review]
Looks good, added a couple of comments

**Pull request merged**